lib/programmemory.cpp

```diff
@@ -1310,6 +1310,7 @@ namespace {
                 return Token::Match(child, "&&|%oror%");
             }))
                 return false;
+            // NOLINTNEXTLINE(bugprone-nondeterministic-pointer-iteration-order) - FP with libc++ only
```

Comment:
I filed https://github.com/llvm/llvm-project/issues/145665 and https://github.com/llvm/llvm-project/issues/145667 about this.